spec/bullet/ext/object_spec.rb

```diff
@@ -38,13 +38,16 @@
       expect(post.bullet_primary_key_value).to eq("#{post.category_id},#{post.writer_id}")
     end
 
-    it 'should return value for multiple primary keys from ActiveRecord 7.1' do
-      allow(Post).to receive(:primary_key).and_return(%i[category_id writer_id])
-      post = Post.first
-      expect(post.bullet_primary_key_value).to eq("#{post.category_id},#{post.writer_id}")
+    if Gem::Version.new(ActiveRecord::VERSION::STRING) >= Gem::Version.new('7.1')
+      it 'should return value for multiple primary keys from ActiveRecord 7.1' do
+        allow(Post).to receive(:primary_key).and_return(%i[category_id writer_id])
+        post = Post.first
```

Comment:
I had to change line 43 and 44 around for memoization so post.bullet_primary_key_value wouldn't reference the old primary_key.
In doing this, we are "forcing" ActiveRecord" to fetch the first post using the composite primary_key instead of the default `id` which fails at some level within AR for older Rails versions.